### PR TITLE
add em channel

### PR DESCRIPTION
**Pull request by harrypuuter**
Added em channel to the configuration

- [x] pair selection algorithm
- [x] trigger setup  
- [x] electron sf added 
- [x] renamed muon sf according to convention

### CODE REVIEW

File: config/config.py [hunk 1]
```diff
@@ -340,19 +346,52 @@ def build_config(
...
+                    "2016postVFP": "data/jsonpog-integration/POG/EGM/2016postVFP_UL/electron.json.gz",
+                    "2017": "data/jsonpog-integration/POG/EGM/2017_UL/electron.json.gz",
+                    "2018": "data/jsonpog-integration/POG/EGM/2018_UL/electron.json.gz",
+                }
+            ),
+            "ele_id_sf_name": "UL-Electron-ID-SF",
+            "ele_sf_year_id": EraModifier(
+                {
+                    "2016preVFP": "2016preVFP",
+                    "2016postVFP": "2016postVFP",
```
Comment (nshadskiy) [inline config/config.py:365]: Is it possible now to split 2016? Then we should also add this for the muon and tau SF configs.

File: config/config.py [hunk 2]
```diff
@@ -340,19 +346,52 @@ def build_config(
...
+                    "2017": "2017",
+                    "2018": "2018",
+                }
+            ),
+            "ele_sf_varation": "sf",  # "sf" is nominal, "sfup"/"sfdown" are up/down variations
+        },
+    )
     # ET/EM channel electron selection
```
Comment (nshadskiy) [inline config/config.py:373]: Typo: only ET channel now

File: code_generation/producers/scalefactors.py [hunk 3]
```diff
@@ -190,3 +190,52 @@
...
+    scopes=["em", "ee", "et"],
+    subproducers={
+        "em": [Ele_1_IDWP90_SF, Ele_1_IDWP80_SF],
+        "ee": [
+            Ele_1_IDWP90_SF,
+            Ele_1_IDWP80_SF,
+            Ele_2_IDWP90_SF,
+            Ele_2_IDWP80_SF,
+        ],
+        "et": [Ele_2_IDWP90_SF, Ele_2_IDWP80_SF],
```
[the remark this answers is not in the document]
Comment (harrypuuter) [reply code_generation/producers/scalefactors.py:239]: yes !

File: config/config.py (hunk 2 above)
Comment (harrypuuter) [reply config/config.py:373]: fixed

**Review by nshadskiy — APPROVED** [review]
Looks good!

**Pull request merged**